core/state-machine/src/backend.rs

```diff
@@ -250,7 +321,8 @@ impl error::Error for Void {
 /// tests.
 pub struct InMemory<H: Hasher> {
 	inner: HashMap<Option<Vec<u8>>, HashMap<Vec<u8>, Vec<u8>>>,
-	trie: Option<TrieBackend<MemoryDB<H>, H>>,
+	kv: Option<InMemoryKvBackend>,
+	state: Option<StateBackend<MemoryDB<H>, H, InMemoryKvBackend>>,
```

Comment:
once you merge master you could renamed it with a more obvious name like as_state_backend_place_holder but I can't really find a good name, but at least the comment